conda_smithy/schema/models.py

```diff
@@ -0,0 +1,476 @@
+        ),
+    ] = {
+        "linux_64": ["azure"],
+        "osx_64": ["azure"],
+        "win_64": ["azure"],
+        # Following platforms are disabled by default
+        "linux_aarch64": None,
+        "linux_ppc64le": None,
+        "linux_armv7l": None,
+        "linux_s390x": None,
+        # Following platforms are aliases of x86_64,
+        "linux": None,
+        "osx": None,
+        "win": None,
+    }
+
+    recipe_dir: str = Field(
+        description="The relative path to the recipe directory",
+        default="recipe",
+    )
+
+    remote_ci_setup: Annotated[
+        Union[List[str], None],
+        Field(
+            description="This option can be used to override the default conda-forge-ci-setup package. Can be given with ${url or channel_alias}::package_name, defaults to conda-forge channel_alias if no prefix is given."
+        ),
+    ] = ["conda-forge-ci-setup=3"]
+
+    shellcheck: Annotated[
+        Union[ShellCheck, None],
+        Field(
+            description="Shell scripts used for builds or activation scripts can be linted with shellcheck. This option can be used to enable shellcheck and configure its behavior."
+        ),
+    ] = None
+
+    skip_render: Annotated[
+        Union[List[str], None],
+        Field(
+            description="This option specifies a list of files which conda smithy will skip rendering. The possible values can be a subset of .gitignore, .gitattributes, README.md, LICENSE.txt. The default value is an empty list [ ]"
+        ),
+    ] = []
+
+    @field_validator("skip_render")
+    def skip_render_validator(cls, v):
+        if not v:
+            return v
+
+        valid_values = [
+            ".gitignore",
+            ".gitattributes",
+            "README.md",
+            "LICENSE.txt",
+            ".github/workflows",
+        ]
+        for i in v:
+            if i not in valid_values:
+                raise ValueError(
+                    f"Invalid value {i} for skip_render. Valid values are {valid_values}"
+                )
+        return v
+
+    templates: Annotated[
+        Union[Dict[str, str], None],
+        Field(
+            description="This is mostly an internal field for specifying where templates files live. You shouldn't need it."
+        ),
+    ] = {}
+
+    test_on_native_only: Annotated[
+        Union[bool, None],
+        Field(
+            description="This is used for disabling testing for cross compiling. Default is false. This has been deprecated in favor of the test top-level field. It is now mapped to test: native_and_emulated."
+        ),
+    ] = False
+
+    test: Annotated[
+        Union[str, None],
+        Field(
+            description="This is used to configure on which platforms a recipe is tested. Default is all. Valid values are all, native_only, native_and_emulated, and emulated_only."
+        ),
+    ] = None
+
+    @field_validator("test")
+    def test_validator(cls, v):
```

Comment:
Same as before. That said, this validator is not needed if the typing uses `Literal[<values allowed>]`.